src/components/email-flow/EmailFlowListProofs.tsx

```diff
@@ -1,25 +1,31 @@
-import { BodyText } from 'components/Text'
+import { BodyText } from 'components/ui/Text'
 import { cursor } from 'classnames/tailwind'
 import { useSnapshot } from 'valtio'
-import EmailProof from 'helpers/EmailProof'
-import ReadyEmailProof from 'components/proofs/ReadyEmailProof'
+import Proof from 'components/proofs/Proof'
+import ProofModel from 'models/Proof'
 import proofStore from 'stores/ProofStore'
 
 export default function ({
   onSelectProof,
 }: {
-  onSelectProof: (proof: EmailProof) => void
+  onSelectProof: (proof: ProofModel) => void
 }) {
-  const { emailProofsCompleted } = useSnapshot(proofStore)
+  const { proofsCompleted } = useSnapshot(proofStore['Email'])
+
   return (
     <>
       <BodyText center>Select a proof to continue</BodyText>
-      {Array.from(emailProofsCompleted).map((proof, index) => (
+      {Array.from(proofsCompleted).map((proof, index) => (
         <div
           onClick={() => onSelectProof(proof)}
           className={cursor('cursor-pointer')}
         >
-          <ReadyEmailProof proof={proof} key={`${proof.domain}-${index}`} />
+          <Proof
+            type="Email"
+            original={proof.original}
```

Comment:
You have a proof, why do you need to pass `original` separately from it? Pass `proof` and get `original` inside the component

Reply:
This component is used in different places in different ways